Mathlib/GroupTheory/Congruence/Basic.lean

```diff
@@ -181,6 +240,31 @@ noncomputable def quotientKerEquivOfSurjective (f : M →* P) (hf : Surjective f
     (ker f).Quotient ≃* P :=
   quotientKerEquivOfRightInverse _ _ hf.hasRightInverse.choose_spec
 
+/-- If e : M →* N is surjective then (c.comap e).Quotient ≃* c.Quotient with c : Con N -/
+@[to_additive "If e : M →* N is surjective then (c.comap e).Quotient ≃* c.Quotient with c :
+AddCon N"]
+noncomputable def comapQuotientEquivOfSurj (c : Con M) (f : N →* M) (hf : Function.Surjective f) :
+    (Con.comap f f.map_mul c).Quotient ≃* c.Quotient :=
+  (Con.congr Con.comap_eq).trans <| Con.quotientKerEquivOfSurjective
+  (c.mk'.comp f) <| Con.mk'_surjective.comp hf
+
+@[to_additive (attr := simp)]
+lemma comapQuotientEquivOfSurj_mk (c : Con M) {f : N →* M} (hf : Function.Surjective f) (x : N) :
+    comapQuotientEquivOfSurj c f hf x = f x := rfl
+
+@[to_additive (attr := simp)]
+lemma comapQuotientEquivOfSurj_symm_mk (c : Con M) {f : N →* M} (hf : Function.Surjective f)
+    (x : N) : (comapQuotientEquivOfSurj c f hf).symm (f x) = x :=
```

Comment:
To show hf can be inferred
```suggestion
lemma comapQuotientEquivOfSurj_symm_mk (c : Con M) {f : N →* M} (hf) (x : N) :
    (comapQuotientEquivOfSurj c f hf).symm (f x) = x :=
```

Reply:
That's cool! Will the user still see the type of hf explicitly in mathlib? I guess my question is, this seems slick, but is it making the lemma harder to use? It may be because I'm a beginner, but I like to know at-a-glance what hypotheses a lemma or theorem requires when I'm on the leanprover-community site

Reply:
No, it's producing exactly the same lemma with exactly the same hover etc